app/src/main/java/com/example/drawingactivity/ColorUtils.kt

```diff
@@ -0,0 +1,74 @@
+package com.example.drawingactivity
+
+import android.graphics.Color
+import java.lang.Integer.parseInt
+
+object ColorUtils {
+
+    /**
+     * A data class that contains color values.
+     * Contains: Hue, Saturation, Value, Red, Green, Blue, and Hex Value.
+     */
+    data class HsvState(
+        var hue: Float,
+        var saturation: Float,
+        var value: Float,
+        var hex: String
+    )
+
+    /**
+     * Calculates the RGB of a color using a HSV float array.
+     * Returns a ColorState containing all values.
+     */
+    fun calculateColorWithHsv(hsv: FloatArray): HsvState {
+        val currentColor: Int = Color.HSVToColor(hsv)
+        val hex = java.lang.String.format(
+            "#%02x%02x%02x",
+            Color.red(currentColor),
+            Color.green(currentColor),
+            Color.blue(currentColor)
+        )
+        return HsvState(
+            hue = hsv[0],
+            saturation = hsv[1],
+            value = hsv[2],
+            hex = hex
+        )
+    }
+
+    /**
+     * Calculates the HSV of a color using a RGB Int array.
+     * Returns a ColorState containing all values.
+     */
+    fun calculateColorWithRgb(rgb: IntArray): HsvState {
```

Comment:
You can make this a static method in Hsv

Reply:
made this a static method